src/transformers/modeling_utils.py

```diff
@@ -148,6 +150,304 @@ def find_tensor_attributes(module: nn.Module) -> List[Tuple[str, Tensor]]:
         return first_tuple[1].dtype
 
 
+def convert_file_size_to_int(size: Union[int, str]):
+    """
+    Converts a size expressed as a string with digits an unit (like `"5MB"`) to an integer (in bytes).
+
+    Args:
+        size (`int` or `str`): The size to convert. Will be directly returned if an `int`.
+
+    Example:
+
+    ```py
+    >>> convert_file_size_to_int("1MB")
+    1048576
+    ```
+    """
+    if isinstance(size, int):
+        return size
+    if size.upper().endswith("GB"):
+        return int(size[:-2]) * (2**30)
+    if size.upper().endswith("MB"):
+        return int(size[:-2]) * (2**20)
+    if size.upper().endswith("KB"):
+        return int(size[:-2]) * (2**10)
+    raise ValueError("`size` is not in a valid format. Use an integer followed by the unit, e.g., '5GB'.")
+
+
+def dtype_size(dtype):
```

Comment:
(nit) since both `bytes` and `bits` is pretty common maybe worth to call it `dtype_byte_size`?